### ISSUE TITLE
Fix false positives for CSS-in-JS in declaration-block-semicolon-newline-after

### ISSUE THREAD

**Issue by dcalhoun** [stylelint/stylelint#5355]
> Clearly describe the bug

There are `declaration-block-semicolon-newline-after` false positives for css-in-js syntax. 

> Which rule, if any, is the bug related to?

`declaration-block-semicolon-newline-after`

> What code is needed to reproduce the bug?

```jsx
const Component = () => (
  <div style={{ color: pink, top: 0 }} />
);
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-semicolon-newline-after": "always"
  }
}
```

> Which version of stylelint are you using?

`13.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.js" --config myconfig.json`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to css-in-js.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
Expected newline after ";" (declaration-block-semicolon-newline-after)
```

**Comment by jeddy3** [comment]
@dcalhoun Thanks for the report and for using the template.

I've labelled as discussion for now as the way syntaxes work [are changing](https://github.com/stylelint/stylelint/issues/5289#issuecomment-840772900) in the [upcoming major release](https://github.com/stylelint/stylelint/issues/5205).

The built-in rules are geared towards standard CSS. I suspect it'll end up being in shared configs that incompatible rules are turned off, rather than in the rules as I don't think they should be aware of syntax (which come and go).

Having said that, there's maybe something we can do with the `isStandardSyntax*` utilities. 

This is one to probably pick up after we release `14.0.0`, and once we see how the adoption of language-specific shared configs that bundle syntaxes goes.

**Comment by jeddy3** [comment]
> I suspect it'll end up being in (shared) configs that incompatible rules are turned off, rather than in the rules as I don't think they should be aware of syntax (which come and go).

This is the approach we intend to take.

You can use the [`overrides`](https://stylelint.io/user-guide/configure#overrides) config property in your own config to turn the rule off for your JavaScript files.
